File: packages/ld-adapter/src/provider/flag-provider.ts
import type {
	ApplyTargetingInput,
	FlagState,
	GetFlagStateInput,
} from '@ff-promo/contracts';

export interface FlagProvider {
	getFlagState(input: GetFlagStateInput): Promise<FlagState>;
	applyTargeting(input: ApplyTargetingInput): Promise<FlagState>;
}

File: kotlin/modules/ld-adapter/src/main/kotlin/com/ffpromo/ldadapter/provider/FlagProvider.kt
package com.ffpromo.ldadapter.provider

import com.ffpromo.contracts.ApplyTargetingInput
import com.ffpromo.contracts.FlagState
import com.ffpromo.contracts.GetFlagStateInput

interface FlagProvider {
    suspend fun getFlagState(input: GetFlagStateInput): FlagState
    suspend fun applyTargeting(input: ApplyTargetingInput): FlagState
}
